p2p/p2p-core/src/client.rs

```diff
@@ -59,8 +60,13 @@ pub struct PeerInformation<A> {
     pub handle: ConnectionHandle,
     /// The direction of this connection (inbound|outbound).
     pub direction: ConnectionDirection,
-    /// The peers pruning seed.
+    /// The peer's [`PruningSeed`].
     pub pruning_seed: PruningSeed,
+    /// The [`CoreSyncData`] of this peer.
+    ///
+    /// Data across fields are not necessarily related, so [`CoreSyncData::top_id`] is not always the
+    /// block hash for the block at height one below [`CoreSyncData::current_height`].
+    pub core_sync_data: Arc<Mutex<CoreSyncData>>,
```

Comment:
If this lock isn't temporary I would really prefer if it were accompanied with a brief description of its scope; how, where, and who the lock should be used by.